lib/known_bugs.pm

```diff
@@ -0,0 +1,48 @@
+#$testapi::distri->set_expected_serial_failures(
+#    soft=>{
+#        "AWESOME SOFT MSG 1"=>[qr/gcc version/], 
+#        "AWESOME SOFT MSG 2"=>[qr/insmod error: 1/]
+#    }, 
+#    hard=>{"AWESOME HARD MSG"=>[qr/No iBFT detected/]
+#    }
+#);
+
+use base Exporter;
+use Exporter;
+use strict;
+use testapi;
+use Data::Dumper;
+
+our @EXPORT = qw(
+    set_known_failure_detection
+);
+
+sub set_known_failure_detection {
+    diag "we're in known bugs";
+    my $serial_failures = [];
+
+    # To add a known bug simply copy and adapt the following line:
+    # push @$serial_failures, {type => soft/hard, message => 'Errormsg', pattern => quotemeta 'ErrorPattern' }
+    # Detect bsc#1093797 on aarch64
+    
+    if (is_sle('=12-SP4') && check_var('ARCH', 'aarch64')) {
+        push @$serial_failures, {type => 'hard', message => 'bsc#1093797', pattern => quotemeta 'Internal error: Oops: 96000006'};
+    }
+
+    if (is_kernel_test()) {
+        my $type = is_ltp_test() ? 'soft' : 'hard';
+        push @$serial_failures, {type => $type, message => 'Kernel Ooops found',             pattern => quotemeta 'Oops:'};
+        push @$serial_failures, {type => $type, message => 'Kernel BUG found',               pattern => qr/kernel BUG at/i};
+        push @$serial_failures, {type => $type, message => 'WARNING CPU in kernel messages', pattern => quotemeta 'WARNING: CPU'};
+        push @$serial_failures, {type => $type, message => 'Kernel stack is corrupted',      pattern => quotemeta 'stack-protector: Kernel stack is corrupted'};
+        push @$serial_failures, {type => $type, message => 'Kernel BUG found',               pattern => quotemeta 'BUG: failure at'};
+        push @$serial_failures, {type => $type, message => 'Kernel Ooops found',             pattern => quotemeta '-[ cut here ]-'};
+    }
+
+    push @$serial_failures, {type => 'hard', message => 'welcome screen found', pattern => quotemeta 'Welcome'};
+    diag Dumper(\$serial_failures);
+    $testapi::distri->set_expected_serial_failures(\$serial_failures);
```

Comment:
Careful here, set_expected_serial_failures is expecting an arrayref, and by passing `\$serial_failures` you will end up passing a [reference to a reference](https://perlmaven.com/ref) (caught me unguarded too :P)